Setup GitHub Actions

mtsmfm opened the pull request:
Feel free to close this PR if you're migrating to https://travis-ci.com.

-----

I've noticed travis-ci.org was ceased two weeks ago.
![image](https://user-images.githubusercontent.com/1796864/123638301-fc35b800-d859-11eb-824a-d7be133dbff6.png)

As a result, currently CI against new PRs like https://github.com/resque/resque/pull/1751 doesn't work.

I confirmed this setting works well:
https://github.com/mtsmfm/resque/pull/1

iloveitaly commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,28 @@
+          ruby-version: "${{ matrix.ruby-version }}"
+      - run: sudo apt-get install redis-server
+      - run: bundle install
+      - run: bundle exec rubocop
```
Could we run rubocop only on a single build?

iloveitaly commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,28 @@
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
```
I think we want to run GH actions on every PR and any branch. No reason not to as far as I'm aware.

```suggestion
on: [push, pull_request]
```

mtsmfm replied:
If we specify both and resque commiters creates new branch on resque/resque repo, both push and pull_request CI run.
Is it ok?

![image](https://user-images.githubusercontent.com/1796864/124525524-91c4df00-de3a-11eb-9fd1-1280c20a9407.png)

iloveitaly replied:
Ah, wasn't aware that the build is run twice in that case. I think it's helpful to developers to run CI against their branch before opening a PR, but probably makes sense to use the default here instead. Thanks for calling this out—feel free to change it back.

iloveitaly commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,28 @@
+          - 2.5.9
+          - 2.6.7
+          - 2.7.3
+          - 3.0.1
```
I think it's better here to drop the x.x.y and just specify x.x so the latest version is always used.

iloveitaly reviewed (changes requested):
This is great! Thank you for your work here.

Left a couple of comments. Could you also remove the travis yaml and remove the travis badge from the readme?

mtsmfm commented on README.markdown:
```diff
@@ -2,7 +2,7 @@ Resque
 ======
 
 [![Gem Version](https://badge.fury.io/rb/resque.svg)](https://rubygems.org/gems/resque)
-[![Build Status](https://travis-ci.org/resque/resque.svg)](https://travis-ci.org/resque/resque)
```
It'll be like:

[![Build Status](https://github.com/mtsmfm/resque/actions/workflows/ci.yml/badge.svg)](https://github.com/mtsmfm/resque/actions/workflows/ci.yml)

iloveitaly replied:
That's great! Thanks for adding the GH badge.

mtsmfm commented:
@iloveitaly Thank you for reviewing.
I fixed ci.yml, removed travis.yml and migrated status badge to GitHub Actions' one.

But I'm not sure if this `on` setting is what you intended.
https://github.com/resque/resque/pull/1752#discussion_r664163306

Can I ask you to review again?

iloveitaly reviewed (changes requested):
You were right about `on`, could you revert that back to the previous config? Otherwise, this looks great! Thank you.

mtsmfm commented:
@iloveitaly Reverted. Can you review again? Thanks!
